std/functional.d

```diff
@@ -260,6 +208,133 @@ unittest
     static assert(!is(typeof(binaryFun!FuncObj)));
 }
 
+/**
+Transforms a string representing an expression into a n-ary function.
+$(UL
+    $(LI The number of parameters is given by `argc`. If no explicit
+        `argc` is given, it is inferred to be `paramNames.length`.)
+    $(LI Explicit names for the function parameters may be given as
+        `paramNames`. If more than `argc` names are supplied, the
+        extras will be ignored.)
+    $(LI The letters (`a` through `z`) are used as default names
+        for up to the first 26 function parameters.)
+    $(LI `args[index]` can always be used to access any parameter by index,
+        regardless of whether it has been given a name.))
+
+If `fun` is not a `string`,  `stringLambda` aliases itself to `fun`.
+In such cases, `stringLambda` does $(B not) verify that `fun` accepts
+the specified number of arguments. (Try $(XREF traits, arity) instead.)
+*/
+template stringLambda(alias fun, size_t argc, paramNames...)
+    if (paramNames.length == 0 || allSatisfy!(isSomeString, typeof(paramNames)))
```

Comment:
> argc <= 100 because __ctfeMatchNary has complexity O(N^2)` (and many other CTFE reasons)
> Does __ctfeMatchNary need to be O(N^2)?
> 
> I haven't studied it enough to figure out exactly what it's doing, but it appears to be a crude lexer / parser type algorithm, which should probably be O(N log(N)) or something.

Yeah... we just need to write it without importing another Phobos and with tiny number of allocation. In addition, it should be fast as previous one for one and two arguments. =P